src/impl_windows.rs

```diff
@@ -43,11 +39,49 @@ pub fn runas_impl(cmd: &Command) -> io::Result<ExitStatus> {
         .collect::<Vec<_>>();
 
     unsafe {
-        let show = if cmd.hide { 0 } else { 1 };
-        Ok(mem::transmute(rust_win_runas(
-            file.as_ptr(),
-            params.as_ptr(),
-            show,
-        )))
+        use winapi::um::combaseapi::CoInitializeEx;
+        use winapi::um::objbase::{COINIT_APARTMENTTHREADED, COINIT_DISABLE_OLE1DDE};
+        use winapi::um::shellapi::{SHELLEXECUTEINFOW, ShellExecuteExW, SEE_MASK_NOASYNC, SEE_MASK_NOCLOSEPROCESS};
+        use winapi::um::winbase::INFINITE;
+        use winapi::um::winuser::{SW_HIDE, SW_NORMAL};
+        use winapi::shared::minwindef::{DWORD, FALSE};
+        use winapi::um::synchapi::WaitForSingleObject;
+        use winapi::um::processthreadsapi::GetExitCodeProcess;
+        use std::ptr;
+
+        let show = if cmd.hide { SW_HIDE } else { SW_NORMAL };
+
+        let mut code : DWORD = 0;
+        let mut sei = SHELLEXECUTEINFOW { 
+            cbSize: mem::size_of::<SHELLEXECUTEINFOW>() as u32,
+            fMask: SEE_MASK_NOASYNC | SEE_MASK_NOCLOSEPROCESS,
+            lpVerb: (*"runas").encode_utf16().collect::<Vec<u16>>().as_ptr(),
```

Comment:
General note: This can also just be a `ptr::null()`, that way it will work for Installers and other binaries. I assume runas was chosen specifically for shell operations?

Reply:
runas is used becuase that is what tells Windows to elevate the process. Otherwise, the process is just launched with non-elevated status.

Here, runas is missing the null-terminator. I believe using `"*runas\0"` would work instead.

Reply:
Changed it to `*"runas\0"`